### PR TITLE
Add support for RSpec 3.5

### PR DESCRIPTION
**Pull request by laserlemon**
Awesome!

### CODE REVIEW

**Comment by bertBruynooghe** [comment]
Awesome indeed! I'll retract my PRs, as you definitely seem to know better what you're doing than I am ;)

File: .travis.yml [hunk 1]
```diff
@@ -29,5 +30,6 @@ rvm:
   - "2.0"
   - "2.1"
   - "2.2"
+  - "2.3"
```
Comment (bertBruynooghe) [inline .travis.yml]: FWIW: didn't got it working like this; had to replace it with ```- 2.3.0``` to have it working.

**Comment by laserlemon** [comment]
I still need to figure out what's preventing the Ruby 1.9.3 build. I don't want to remove support just because Travis is wonky. Also, the Ruby 2.3 builds are broken and I think it's because of the version of RVM that Travis has installed. If you can help track down solutions for the failing builds, that would be fantastic!

**Comment by bertBruynooghe** [comment]
As for the 2.3 builds, I do have an issue on my local machine using rbenv 2.3.1; the issue you had on travis can be remediated by putting - rvm: 2.3.0 instead, but then you’ll have the following issue:

wait for not_to accepts a value rather than a block
     Failure/Error: expect {
       expected no Exception, got #<NoMethodError: private method `wait_timeout' called for #<RSpec::Core::Configuration:0x000000029670b0>>
     # ./spec/wait_spec.rb:170:in `block (4 levels) in <top (required)>'
     # ./lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>’

The 1.9.3 issue seems like a version of json that is too high causing bundle to fail: I spend some time finding out what went wrong, but your conclusion that it just a Travis issue seems to fall short: if you try to bundle the gemfile locally, you get exactly the same issues.

I’ll see if I can get further on any of the issues…

**Comment by bertBruynooghe** [comment]
1.9 issues seems like a bundler issue; if I add:

    gem 'json', '~> 1.8'

to the spec_2_11.gemfile, I can pass ```BUNDLE_GEMFILE=$PWD/gemfiles/rspec_2_11.gemfile bundle update```  and ```... bundle exec spec``` on ruby 1.9.3-p551 locally.

**Comment by laserlemon** [comment]
@bertBruynooghe After struggling with this for a while, I conceded that we could (and probably should) drop support for Ruby 1.9 and 2.0; they're both end-of-life. So that "solves" the Ruby 1.9 issue.

As for the Ruby 2.3 issue, it seems as though RSpec versions < 2.14 use `eval` for custom configuration options (like `wait_timeout` and `wait_delay`). It seems as though Ruby 2.3 changes the behavior of `eval` so that methods defined within are private rather than public. There are some hacky solutions that can make it work but the gist is that RSpec 2.13 was never (and still isn't) meant to work with Ruby 2.3. Rather than fight it, I chose to drop support for RSpec 2. The latest version was 2.14.1 which was released three years ago.

If RSpec 2 support is still needed, RSpec::Wait version 0.0.8 will still work. The only issue I can imagine is if you wanted to combine RSpec 2 and Rails 5, and that is such a crazy outside possibility.

**Comment by bertBruynooghe** [comment]
Thank you very much.
Do you have any plans on releasing soon?

**Comment by laserlemon** [comment]
👍

**Pull request merged**